src/base.ts

```diff
@@ -656,7 +793,15 @@ export default class KintoClientBase {
    * @param  {Number}        [options.last_modified] The last_modified option.
    * @return {Promise<Object, Error>}
    */
-  async deleteBucket(bucket, options = {}) {
+  async deleteBucket(
+    bucket: string | KintoIdRecord,
```

Comment:
If this «id record» type has to be used in operations on buckets, then we should call it `KintoIdObject`